src/Cassandra/Mapping/CqlBatch.cs

```diff
@@ -77,7 +77,7 @@ private void Insert<T>(bool ifNotExists, bool insertNulls, T poco, CqlQueryOptio
             var values = getBindValues(poco);
             //generate INSERT query based on null values (if insertNulls set)
             object[] queryParameters;
-            var cql = _cqlGenerator.GenerateInsert<T>(insertNulls, values, out queryParameters, ifNotExists, ttl);
+            var cql = _cqlGenerator.GenerateInsert<T>(insertNulls, values, out queryParameters, ifNotExists, ttl, queryOptions?.Timestamp);
```

Comment:
Null-conditional operator is a C# 6 language feature. Can you replace it with conditional operator?

We are still using C# 4.0, [will start using C# 5 soon](https://datastax-oss.atlassian.net/browse/CSHARP-426) but using C# 6.0 means requiring newer compiler versions (dropping msbuild 12).